### PR TITLE
Refactor web-ci pipeline and delete eager mode CI pipeline

### PR DESCRIPTION
**Pull request by snnn**
### Description
1. Move it to a separated pool that use the same image as [the public hosted pool](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml). Also, create a beta pool which contains the next version image of the hosted pool, and add jobs in our post merge pipeline to test if the next version image will break our CI. So, usually we will have at least one week to prepare. 

2. Change the cmake generator in use in our pipelines from "Ninja" to "MingW Makefile", because the latest version of cmake doesn't work with the latest version of Ninja.  People who prefer Ninja could still use ninja in their local build by passing "--cmake_generator ninja" to [build.py](https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/build.py).

3. Delete eager mode CI pipeline.  Confirmed with @souptc .

### Motivation and Context
I need to update the software we have in our CI build machines, and I need to resolve this incompatibility issue.  In more detail, the build error I hit was:

em++: error: CMakeFilesonnxruntime_mlas_test.dirC_a_work1sonnxruntimetestmlasunittesttest_activation.cpp.o: No such file or directory ("CMakeFilesonnxruntime_mlas_test.dirC_a_work1sonnxruntimetestmlasunittesttest_activation.cpp.o" was expected to be an input file, based on the commandline arguments provided)

After this PR we will deprecate python 3.7 support. The eager mode CI pipeline is the last one that still use python 3.7. Then we can rework the PR #10953 made by [fs-eire](https://github.com/fs-eire) last year. 

Fixed [AB#14435](https://aiinfra.visualstudio.com/6a833879-cd9b-44a4-a9de-adc2d818f13c/_workitems/edit/14435)

### CODE REVIEW

**Review by souptc — DISMISSED** [review]

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -1738,7 +1729,7 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
         # Adding the torch lib path for loading DLLs for onnxruntime in eager mode
         # This works for Python 3.7 and below, and doesn't work for Python 3.8+
         # User will need to import torch before onnxruntime and it will work for all versions
-        if (args.build_eager_mode or args.enable_lazy_tensor) and is_windows():
+        if (args.enable_lazy_tensor) and is_windows():
```
[the remark this answers is not in the document]
Comment (snnn) [reply tools/ci_build/build.py:1732]: Will do. This file definitely needs more clean-ups.

**Review by fs-eire — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/templates/win-wasm-ci.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
   variables:
     EnvSetupScript: setup_env.bat
     buildArch: x64
-    CommonBuildArgs: '--parallel --config ${{ parameters.BuildConfig }} --skip_submodule_sync --cmake_generator "Visual Studio 16 2019" --build_wasm --use_xnnpack --emsdk_version releases-29ad1037cd6b99e5d8a1bd75bc188c1e9a6fda8d-64bit ${{ parameters.ExtraBuildArgs }}'
+    CommonBuildArgs: '--parallel --config ${{ parameters.BuildConfig }} --skip_submodule_sync --cmake_generator "MinGW Makefiles" --build_wasm --use_xnnpack --emsdk_version releases-29ad1037cd6b99e5d8a1bd75bc188c1e9a6fda8d-64bit ${{ parameters.ExtraBuildArgs }}'
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/templates/win-wasm-ci.yml:32]: Not sure about MinGw, are we going to add the MinGW as a dependency now?

**Review by jchen351 — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/templates/win-wasm-ci.yml (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/win-wasm-ci.yml:32]: Yes. Because ninja stops working.

**Comment by thiagocrepaldi** [comment]
@snnn Why the `--test_external_transformer_example` flag was deleted? As it happens, it seems the `--external_graph_transformer_path` flag is broken and it wasn't detected y CI because this option was removed from `build.py` in a PR that doesn't seemt o be about external graph transformers

**Pull request merged**